### PR TITLE
Add recipe for chemdyg

### PR DESCRIPTION
**Pull request by xylar**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/chemdyg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  license_family: BSD
+  license_file: LICENSE
+  doc_url: https://github.com/E3SM-Project/ChemDyg
+  dev_url: https://github.com/E3SM-Project/ChemDyg
+
+extra:
+  recipe-maintainers:
+    - xylar
+    - hsiangheleellnl
+    - tangq
```
Comment (xylar) [inline recipes/chemdyg/meta.yaml:58]: @hsiangheleellnl and @tangq, please respond to this message stating that you are willing to be maintainers of this conda package (this is required as part of the review of the recipe).

Comment (hsiangheleellnl) [reply recipes/chemdyg/meta.yaml:58]: Yes, I am willing to be a maintainer of the ChemDyg package.

Comment (tangq) [reply recipes/chemdyg/meta.yaml:58]: Me too.

**Comment by xylar** [comment]
@conda-forge/help-python, ready for review!

Windows build failures are expected because dependencies are not available for Windows.

File: recipes/chemdyg/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  commands:
+    - pip check
+
+about:
+  home: https://github.com/E3SM-Project/ChemDyg
+  summary: 'Chemistry Diagnostics Package'
+  description: |
+    E3SM chemistry diagnostic package version1.0 (ChemDyg v1.0) is designed for
+    E3SM chemistry development. There are total 11 types of plot and 3 types of
+    table for model-to-model and model-to-observation comparison.
```
[the remark this answers is not in the document]
Comment (xylar) [reply recipes/chemdyg/meta.yaml:46]: Excellent point, I'll accept your suggetion.

**Comment by xylar** [comment]
Thank you very much @ocefpaf!

**Comment by tangq** [comment]
Thank you so much @ocefpaf

**Pull request merged**